agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/utils/SanitizationHelper.java

```diff
@@ -0,0 +1,136 @@
+      Map<String, Map.Entry<String, String>> sanitizedProperties = new HashMap<>();
+      for (Map.Entry<String, String> entry : properties.entrySet()) {
+        String sanitizedKey = sanitizeKey(entry.getKey());
+        String sanitizedValue = sanitizeValue(entry.getValue());
+        if (Strings.isNullOrEmpty(sanitizedValue)
+            || !sanitizedKey.equals(entry.getKey())
+            || !sanitizedValue.equals(entry.getValue())) {
+          sanitizedProperties.put(entry.getKey(), Pair.of(sanitizedKey, sanitizedValue));
+        }
+      }
+
+      for (Map.Entry<String, Map.Entry<String, String>> entry : sanitizedProperties.entrySet()) {
+        properties.remove(entry.getKey());
+        if (!Strings.isNullOrEmpty(entry.getValue().getValue())) {
+          String uniqueKey = makeKeyUnique(entry.getValue().getKey(), properties);
+          properties.put(uniqueKey, entry.getValue().getValue());
+        }
+      }
+    }
+  }
+
+  /** Function to create unique key. */
+  private static String makeKeyUnique(String key, Map<String, ?> map) {
+    if (map.containsKey(key)) {
+      String truncatedKey = Strings.truncate(key, MAX_KEY_NAME_LENGTH - 3);
+      int candidate = 1;
+      do {
+        key = truncatedKey + candidate;
+        ++candidate;
+      } while (map.containsKey(key));
+    }
+
+    return key;
+  }
+
+  /** Function to sanitize value. */
+  private static String sanitizeValue(String value) {
+    return Strings.trimAndTruncate(value, MAX_VALUE_LENGTH);
+  }
+
+  /** Function to sanitize key. */
+  private static String sanitizeKey(String key) {
+    String sanitizedKey = Strings.trimAndTruncate(key, MAX_KEY_NAME_LENGTH);
+    return makeKeyNonEmpty(sanitizedKey);
+  }
+
+  /** Function to return non empty key. */
+  private static String makeKeyNonEmpty(String sanitizedKey) {
+    // TODO check if this is valid. Below code is based on .Net implementation of the same
+    return Strings.isNullOrEmpty(sanitizedKey) ? "required" : sanitizedKey;
+  }
+
+  /** Function to sanitize key value pair in measurements. */
+  public static void sanitizeMeasurements(Map<String, Double> measurements) {
+    if (measurements != null) {
+      Map<String, Map.Entry<String, Double>> sanitizedMeasurements = new HashMap<>();
+      for (Map.Entry<String, Double> entry : measurements.entrySet()) {
+        String sanitizedKey = sanitizeKey(entry.getKey());
+        // TODO sanitize value ?
+        if (!sanitizedKey.equals(entry.getKey())) {
+          sanitizedMeasurements.put(entry.getKey(), Pair.of(sanitizedKey, entry.getValue()));
+        }
+      }
+      for (Map.Entry<String, Map.Entry<String, Double>> entry : sanitizedMeasurements.entrySet()) {
+        measurements.remove(entry.getKey());
+        String uniqueKey = makeKeyUnique(entry.getValue().getKey(), measurements);
+        measurements.put(uniqueKey, entry.getValue().getValue());
+      }
+    }
+  }
+
+  private SanitizationHelper() {}
+
+  static class Pair {
```

Comment:
I think this class would be more readable as a real class named `SanitizedEntry`, with methods getKey() and getValue()